process/block/metablock.go

```diff
@@ -2032,10 +2151,24 @@ func (mp *metaProcessor) createShardInfo() ([]data.ShardDataHandler, error) {
 		shardData.PrevRandSeed = shardHdr.GetPrevRandSeed()
 		shardData.PubKeysBitmap = shardHdr.GetPubKeysBitmap()
 		if mp.enableEpochsHandler.IsFlagEnabledInEpoch(common.EquivalentMessagesFlag, shardHdr.GetEpoch()) {
-			proof := shardHdr.GetPreviousProof()
-			if proof != nil {
-				shardData.PubKeysBitmap = proof.GetPubKeysBitmap()
+			prevProof := shardHdr.GetPreviousProof()
+			if prevProof != nil {
+				shardData.PubKeysBitmap = prevProof.GetPubKeysBitmap()
+				shardData.Signature = prevProof.GetAggregatedSignature()
+				err := shardData.SetPreviousProof(prevProof)
```

Comment:
why do we set here pubkeysbitmap and signature from previous proof?

Reply:
because now the bitmap + signature from block are part of the proof of the previous block